fs-v1-multipart.go

```diff
@@ -322,10 +324,17 @@ func (fs fsObjects) PutObjectPart(bucket, object, uploadID string, partID int, s
 	newMD5Hex := hex.EncodeToString(md5Writer.Sum(nil))
 	if md5Hex != "" {
 		if newMD5Hex != md5Hex {
+			fs.storage.DeleteFile(minioMetaBucket, tmpPartPath)
 			return "", BadDigest{md5Hex, newMD5Hex}
 		}
 	}
 
+	if size != bytesRead {
+		// content length doesn't match with part size, delete the part
+		fs.storage.DeleteFile(minioMetaBucket, tmpPartPath)
+		return "", errSignatureMismatch
```

Comment:
Returning this error even in case of 'anonymous' requests is going to be wrong and unexpected which will happen since there is not real validation should this be returned in such a case. 

If you look at the flow the signature errors are not returned from this point on-wards.  Signature verification happens at object-handlers through a io.Pipe(). The reason to do that was signature verification should happen outside of the object API - since all the HTTP related details are not spilled over into this layer.